api/objects.proto

```diff
@@ -222,3 +222,34 @@ message Cluster {
 	// a new key is allocated on key rotation.
 	uint64 encryption_key_lamport_clock = 6;
 }
+
+// Secret
+message Secret {
+	string id = 1 [(gogoproto.customname) = "ID"];
+
+	Meta meta = 2 [(gogoproto.nullable) = false];
+
+	map<string, SecretData> secret_data = 3;
+
+	// Name of the secret, so that we can use it easily for lookups
+	string name = 4;
+
+	// ID of the latest version, so that we can easily use it for lookups
+	string latest = 5 [(gogoproto.customname) = "LatestVersion"];
+}
+
+// SecretData is represents a specific version of a secret, and is immutable
+message SecretData {
+	string id = 1 [(gogoproto.customname) = "ID"];
+
+	Meta meta = 2 [(gogoproto.nullable) = false];
```

Comment:
Yes, I think it should just be a timestamp.

`Meta` is meant to be used in top-level objects and is automatically used by the object store. Right now it contains three fields:

- `Version`: used by the store for object versioning. This presumably wouldn't be used in `SecretData` because `SecretData` values are immutable (right?)
- `CreatedAt`: creation timestamp
- `UpdatedAt`: modification timestamp. Assuming `SecretData` values are immutable, this one wouldn't make sense either.